### ISSUE TITLE
Nested comments should understand quoted strings

### ISSUE THREAD

**Issue by jfehrle** [reasonml-editor/reasonml-idea-plugin#26]
Comments should ignore "(\*" or "\*)" in quoted strings.  They should not begin or end comments.
 
![image](https://user-images.githubusercontent.com/1253341/34656113-eb23127e-f3c9-11e7-9bee-021dff51bc33.png)

```
$ ocaml
        OCaml version 4.04.2
# (* "(*" *);;
# (* "*)" *);;
# (* *) *);;
Characters 6-8:
  (* *) *);;
        ^^
Warning: this is not the end of a comment.
Characters 6-7:
  (* *) *);;
        ^
Error: Syntax error
```

**Comment by giraud** [comment]
Why would you want to do that ? 
I don't see a value to add complexity to the lexer for that use case.
For example, in java, you get the same result: the comment is closed at the first */ found in a string.

**Comment by jfehrle** [comment]
I ran into this while I was working on my own program.  I commented out a large block of code to debug something and got confused because the editor and the OCaml compiler (not Reason) didn't agree on what was commented out.  Took me a while to figure out.

OCaml supports nested comments and understands this case.  I think my exact case was commenting out something like this code, which incorrectly was interpreted as a new nesting level.

```
my_func "(*" "another arg"
```

Of course, your choice to decide what to support and when.

Regards.
